### PR TITLE
Decoder improvements

### PR DESCRIPTION
**Pull request by ethanacm**
Various improvements to decoding, mostly as listed on [this trello board](https://trello.com/b/W6GV8Qch/decoder-polishing).

### CODE REVIEW

File: importers/importer.py [hunk 1]
```diff
@@ -100,21 +100,20 @@ def import_point(points):
...
-                continue
+        for (table, id, tags) in [('buildings', building_id, point.get_building_tags()),
+                                  ('rooms', room_id, point.get_room_tags()),
+                                  ('devices', device_id, point.get_device_tags()),
+                                  ('points', point_id, point.get_point_tags())]:
             for tag in tags:
-                tag_id = get_id_of("tags", tag)
-                if tag_id is None:
-                    execute_and_commit("""INSERT INTO tags (name) VALUES (%s)""", (tag,))
+                if tag is not None:
```
Comment (alextdavis) [inline importers/importer.py:108]: Wouldn't it be safe to assume that `tags` will be an array of non-None values? Not necessarily saying we need to change this, but just want to confirm my understanding.

**Review by alextdavis — APPROVED** [review]
Everything here looks good to me!

File: importers/importer.py (hunk 1 above)
Comment (grenche) [reply importers/importer.py:108]: I could be wrong, but I think right now, the ____type (device_type for example) is always going in the list of tags regardless of whether or not it is None. I guess we could decide to only add tags to the list that are non-None values or check it here. I'm not sure which I prefer.

**Review by grenche — APPROVED** [review]
I think there is maybe still a little more to think about with importing tags, but other than that everything looks good! I do think we need more testing given how many errors we've been finding.

File: importers/importer.py (hunk 1 above)
Comment (ethanacm) [reply importers/importer.py:108]: Yup! I agree it doesn't make sense to ever add none to the tags list. Fixed in a new commit, please give a super quick look and approve when you can :)

**Pull request merged**